src/push-manifest.ts

```diff
@@ -0,0 +1,163 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {writeFile} from 'fs';
+import * as path from 'path';
+import {Analyzer, FSUrlLoader} from 'polymer-analyzer';
+
+import {urlFromPath} from './path-transformers';
+import {PolymerProject} from './polymer-project';
+
+
+export interface AddPushManifestOptions {
+  project: PolymerProject;
+  buildRoot: string;
+  bundled?: boolean;
+  path?: string;
+}
+
+// Push Manifest Types Definitions
+export type LinkType = 'document' | 'script' | 'style' | 'image' | 'font';
+const extensionToTypeMapping: {[extname: string]: LinkType} = {
+  '.css': 'style',
+  '.gif': 'image',
+  '.html': 'document',
+  '.png': 'image',
+  '.jpg': 'image',
+  '.js': 'script',
+  '.json': 'script',
+  '.svg': 'image',
+  '.webp': 'image',
+  '.woff': 'font',
+  '.woff2': 'font'
+};
+
+
+export interface PushManifestEntry {
+  [id: string]: {
+    type?: LinkType,
+    weight: 1,
+  };
+}
+export interface PushManifest { [id: string]: PushManifestEntry; }
```

Comment:
Why is there an id in both `PushManifestEntry` and `PushManifest`? How are they different, what are they id's of?